src/components/grapher/GrapherTool.vue

```diff
@@ -1,8 +1,8 @@
 <template>
   <g class="grapher-tool--tool-dots-container">
     <circle
-      v-for="dot in grapherToolDots"
-      :key="`${dot.x}-${dot.y}-tool--dot`"
+      v-for="(dot, index) in grapherToolDots"
+      :key="`${dot.x}-${dot.y}-${index}-tool--dot`"
```

Comment:
Should key be:
`${index}-tool--dot`
?

Reply:
Good call-out! In React I'm used to making `key` not rely solely on `index` just in case of edge cases where it does not work efficiently... If you're interested you can see the React docs... They provide an article with more explanation. https://reactjs.org/docs/lists-and-keys.html#keys

In this case, I don't think we need to worry about `grapherToolDots` being re-ordered, which is one of the reasons to avoid using `index` in React.